go-bindings/threshold.go

```diff
@@ -0,0 +1,224 @@
+// Copyright (c) 2021 The Dash Core developers
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//    http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package blschia
+
+// #cgo LDFLAGS: -lbls-dash
+// #cgo CXXFLAGS: -std=c++14
+// #include <stdlib.h>
+// #include "threshold.h"
+// #include "blschia.h"
+import "C"
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"errors"
+	"runtime"
+	"unsafe"
+)
+
+// HashSize is the allowed size of a hash
+const HashSize = 32
+
+// Hash represents 32 byte of hash data
+type Hash [32]byte
+
+var errWrongHexStringValue = errors.New("a hex string must contain 32 bytes")
+
+// HashFromString convert a hex string into a Hash
+func HashFromString(hexString string) (Hash, error) {
+	var hash Hash
+	data, err := hex.DecodeString(hexString)
+	if err != nil {
+		return hash, err
+	}
+	if len(data) < HashSize {
+		return hash, errWrongHexStringValue
+	}
+	for i, d := range data[len(data)-HashSize:] {
+		hash[HashSize-(i+1)] = d
+	}
+	return hash, nil
+}
+
+// BuildSignHash creates the required signHash for LLMQ threshold signing process
+func BuildSignHash(llmqType uint8, quorumHash Hash, signID Hash, msgHash Hash) Hash {
+	hasher := sha256.New()
+	hasher.Write([]byte{llmqType})
+	hasher.Write(quorumHash[:])
+	hasher.Write(signID[:])
+	hasher.Write(msgHash[:])
+	return sha256.Sum256(hasher.Sum(nil))
```

Comment:
Is it correct? If I understand correctly, you are doing a sha256 with Sum() first, and then another sha256 from received checksum.
It's double hashing.

I guess you should just return `hasher.Sum(nil)`

Reply:
as I mentioned above, I copied this function from [here](https://github.com/dashpay/bls-signatures/blob/develop_0.1/go-bindings/util.go#L30)

Reply:
@QuantumExplorer  your opinion?

Reply:
Core uses a double sha, this is correct.